common/FileUtils.java

```diff
@@ -102,46 +98,48 @@ public void writeStringUnchecked(String text, String path) {
 
   /** Checks if file exists. Returns false for folders. */
   public boolean fileExists(String path) {
-    File file = new File(expandHomeDirectory(path));
-    return file.exists() && file.isFile();
+    return Files.isRegularFile(fileSystem.getPath(expandHomeDirectory(path)));
   }
 
   /** Checks if folder exists. Returns false for files. */
   public boolean folderExists(String path) {
-    return java.nio.file.Files.isDirectory(fileSystem.getPath(expandHomeDirectory(path)));
+    return Files.isDirectory(fileSystem.getPath(expandHomeDirectory(path)));
   }
 
   /** Checks if folder or folder exists. */
   public boolean fileOrFolderExists(String path) {
-    File file = new File(expandHomeDirectory(path));
-    return file.exists();
+    return Files.isRegularFile(fileSystem.getPath(expandHomeDirectory(path)))
+            || Files.isDirectory(fileSystem.getPath(expandHomeDirectory(path)));
   }
 
   /** Creates directories in path if none exist. */
   public void mkdirs(String path) {
     try {
-      Files.createParentDirs(new File(path));
+      Files.createDirectories(fileSystem.getPath(expandHomeDirectory(path)));
     } catch (IOException e) {
       throw new RuntimeException(e);
     }
   }
 
   /** Gets filenames in path. */
-  public ImmutableList<String> getFiles(String path) {
-    String[] files = Paths.get(path).toFile().list();
-    if (files.length == 0) {
-      return ImmutableList.of();
+  public ImmutableList<String> getFiles(String path) throws IOException {
+    List<String> files = new ArrayList<>();
+    try (Stream<Path> paths = Files.list(fileSystem.getPath(expandHomeDirectory(path)))) {
```

Comment:
@oferb AFAICT, the problem of doing it that way as a one-liner is that `Files.list` returns a stream that must be closed, because otherwise it doesn't close the underlying file on the OS side. So I'd suggest something like this:

```java
try (Stream<Path> paths = Files.list(fileSystem.getPath(expandHomeDirectory(path)))) {
  ... = paths.map(...)...
}
```

Reply:
Yay!
I love the internet. Thank you!

Reply:
@oferb You're very welcome! :)